public/resources/core/js/controllers/customer.controller.js

```diff
@@ -62,14 +63,27 @@
                 });
             });
         });
-
-        Customer.search().then(function (customers) {
-            vm.list = customers;
-            vm.status = DATA_STATUS.LOADED;
-        }, function (response) {
-            vm.errorMessage = Utils.parseError(response, 'Unexpected error trying to retrieve the list of shipping address.');
-            vm.status = DATA_STATUS.ERROR;
-        });
+	
+	$scope.$on(Party.EVENTS.USER_SESSION_SWITCHED, function (event, message, obj) {
+	    updateShippingList();
+	});
+
+	function canEditOrAdd() {
+	    return User.loggedUser.currentUser.id === User.loggedUser.id || Party.hasAdminRole();
+	};
+
+	updateShippingList();
+	
+	function updateShippingList() {
+            Customer.search().then(function (customers) {
```

Comment:
Check if this method is retrieving all the customer objects of the system (have look also at the controller in the backend). In that case, modify waht needed so you can retrieve only what you need